GANDLF/metrics/classification.py

```diff
@@ -25,6 +23,36 @@ def overall_stats(
         params["problem_type"] == "classification"
     ), "Only classification is supported for these stats"
 
+    def __convert_tensor_to_int(input_tensor: torch.Tensor) -> torch.Tensor:
+        """
+        Convert the input tensor to integer format.
+
+        Args:
+            input_tensor (torch.Tensor): The input tensor.
+
+        Returns:
+            torch.Tensor: The tensor converted to integer format.
+        """
+        return_tensor = input_tensor
+        if return_tensor.dtype != torch.long or return_tensor.dtype != torch.int:
+            return_tensor = return_tensor.long()
+        return return_tensor
+
+    # this is needed for a few metrics
+    # ensure that predictions and target are in integer format
+    prediction_wrap = __convert_tensor_to_int(prediction)
+    target_wrap = __convert_tensor_to_int(target)
+
+    # this is needed for auroc
+    # ensure that predictions are in integer format
```

Comment:
Is it checked somehow? I believe this is not necessairly true regarding the preds being int format. The idea of auroc itself is to check the model true and false positive rates in different decision thresholds, therefore any thresholding is possible only on logits, otherwise it does not make sense. User guide of torchmetrics also show example of raw logits https://lightning.ai/docs/torchmetrics/stable/classification/auroc.html

Reply:
I needed to have this in place to get auroc to work for the BraTS-Path metrics (#885). Let's open up a separate issue to check this, instead?

Reply:
Yeah, good idea.